samples/sensor/ltr390/src/main.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2022, Luke Holt
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/sensor.h>
+
+#include <stdio.h>
+
+#ifdef CONFIG_LTR390_TRIGGER
+
+#define INITIAL_LIGHT_LEVEL     100.0
+#define LIGHT_LEVEL_VARIANCE    3
+
+static struct sensor_trigger trigger;
+
+static int set_window(const struct device *dev,
+					  enum sensor_channel chan,
+					  const double light_level)
+{
+	int rc;
+
+	struct sensor_value lower, upper;
+
+	rc = sensor_value_from_double(&lower, light_level - LIGHT_LEVEL_VARIANCE);
+	if (rc < 0) {
+		printf("Could not convert double to value struct");
+		return rc;
+	}
+	rc = sensor_value_from_double(&upper, light_level + LIGHT_LEVEL_VARIANCE);
+	if (rc < 0) {
+		printf("Could not convert double to value struct");
+		return rc;
+	}
+
+	rc = sensor_attr_set(dev, chan, SENSOR_ATTR_LOWER_THRESH, &lower);
+	if (rc < 0) {
+		printf("Could not set lower threshold");
+		return rc;
+	}
+
+	rc = sensor_attr_set(dev, chan, SENSOR_ATTR_UPPER_THRESH, &upper);
+	if (rc < 0) {
+		printf("Could not set lower threshold");
+		return rc;
+	}
+
+	printf("Alert on ambient light [%d.%06d, %d.%06d] lux\n",
+		lower.val1, lower.val2, upper.val1, upper.val2);
+
+	return 0;
+}
+
+static void trigger_handler(const struct device *dev,
+							const struct sensor_trigger *trig)
+{
+	struct sensor_value val;
+	static size_t cnt;
+	int rc;
+
+	++cnt;
+	rc = sensor_sample_fetch(dev);
+	if (rc < 0) {
+		printf("sensor_sample_fetch error: %d\n", rc);
+		return;
+	}
+
+	rc = sensor_channel_get(dev, trig->chan, &val);
+	if (rc < 0) {
+		printf("sensor_channel_get error: %d\n", rc);
+		return;
+	}
+
+	printf("Trigger fired %u, val: %d.%06d\n", cnt, val.val1, val.val2);
+
+	set_window(dev, trig->chan, sensor_value_to_double(&val));
+}
+#endif /* CONFIG_LTR390_TRIGGER */
+
+void main(void)
```

Comment:
The project has been trying hard to avoid each sensor having its own sample but instead having a common sample application covering each sensor type.

It would be great if this sample could be reworked to do just that for light_polling sample akin to the accel_polling and magn_polling samples in the tree today.

It would probably be ok for the sensor to indicated it has specific features if those are needed in the sample, e.g. having a threshold attribute.

Reply:
Should I include the UVI channel in the light_polling sample? The accel_polling and magn_polling samples target `ACCEL_(X|Y|Z)` and `MAGN(X|Y|Z)` respectively.

If not, should I create a uvi_polling sample?

Reply:
That makes sense to me, @MaureenHelm any thoughts?

Reply:
I added a light_polling sample.

I left the ltr390 sample, because otherwise, there would be no demonstration of the trigger functionality. Should this instead be added to light_polling?

Any thoughts?